Framework/MDAlgorithms/src/MDNormSCD.cpp

```diff
@@ -363,6 +363,16 @@ void MDNormSCD::cacheDimensionXValues() {
   }
 }
 
+namespace {
+template <typename T, typename BinaryOp>
+void AtomicOp(std::atomic<T> &f, T d, BinaryOp _Op) {
+  T old = f.load();
+  T desired = _Op(old, d);
+  while (!f.compare_exchange_weak(old, desired))
+    desired = _Op(old, d);
```

Comment:
It's possible that between lines 369 and 371 `f` was updated by another thread. The while loop updates the value of `desired` and tries again.